File: 단계별로 풀어보기/문자열/1316/그룹 단어 체커.py
def groupWordChecker(word):
    alphabet= set()
    alphabet.add(word[0])
    for i in range(1,len(word)):
        if (word[i]!=word[i-1]) and (word[i] in alphabet):
            return False
        else:
            alphabet.add(word[i])
    return True
    
def init():
    N=int(input())
    count=0
    for i in range(N):
        word=list(input())
        if(groupWordChecker(word)):
            count+=1
    print(count)
    
    return 0;
    
init()

File: 단계별로 풀어보기/문자열/1316/그룹 단어 체커.cpp
#include <iostream>
#include <set>
using namespace std;

bool groupWordChecker(char* word){
    set<char> alphabet;
    alphabet.insert(word[0]); //set에는 push_back()없고 insert()씀
    
    for(int i=1; word[i]!='\0'; i++){
        if((word[i-1]!=word[i])&&(alphabet.find(word[i])!=alphabet.end()))
            return false;
        else
            alphabet.insert(word[i]);
    }
    
    return true;
}

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int N;
    int count=0;
    char word[101];
    cin >> N;
    
    for(int i=0; i<N; i++){
        cin >> word;
        if(groupWordChecker(word))
            count++;
    }
    
    cout<<count;
    return 0;
}
